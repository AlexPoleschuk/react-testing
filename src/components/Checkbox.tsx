import {
    type FC,
    useState,
} from 'react';

interface CheckboxPropsType {
    isChecked?: boolean;
}

const Checkbox: FC<CheckboxPropsType> = ({
    isChecked = false,
}) => {
    const [checked, setChecked] = useState(isChecked);

    const onChange = () => {
        setChecked(!isChecked);
    };

    return (
        <input
            type="checkbox"
            role="test_checkbox"
            checked={checked}
            onChange={onChange}
        />
    );
};

export default Checkbox;
