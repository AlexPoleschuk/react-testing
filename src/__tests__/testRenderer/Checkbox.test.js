import { create, act } from 'react-test-renderer';

import Checkbox from '../../components/Checkbox';

describe('Checkbox', () => {
    test('1) is component rendered', () => {
        const component = create(
            <Checkbox />
        );

        const testInstance = component.root;

        expect(testInstance.findByType('input').props.checked).toBe(false);
    });

    test('2) making snapshots with checked/unchecked', () => {
        const component = create(
            <Checkbox />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        act(() => {
            tree.props.onChange();
        });

        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
