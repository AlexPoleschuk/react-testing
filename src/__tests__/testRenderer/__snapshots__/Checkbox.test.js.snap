// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox 2) making snapshots with checked/unchecked 1`] = `
<input
  checked={false}
  onChange={[Function]}
  role="test_checkbox"
  type="checkbox"
/>
`;

exports[`Checkbox 2) making snapshots with checked/unchecked 2`] = `
<input
  checked={true}
  onChange={[Function]}
  role="test_checkbox"
  type="checkbox"
/>
`;
